src/dfx-core/src/error/load_tool_config.rs

```diff
@@ -0,0 +1,15 @@
+use super::{config::ConfigError, structured_file::StructuredFileError};
+
+use thiserror::Error;
+
+#[derive(Error, Debug)]
+pub enum ToolConfigError {
+    #[error("Failed to get path for tool configuration")]
+    GetConfigPathFailed(#[source] ConfigError),
+
+    #[error("Failed to load tool configuration")]
+    LoadConfigFromFileFailed(#[source] StructuredFileError),
+
+    #[error("Failed to save default tool configuration")]
```

Comment:
When called from update_config, this isn't saving the default tool configuration.
```suggestion
    #[error("Failed to save tool configuration")]
```